app/src/main/java/com/bnyro/recorder/ui/models/PlayerModel.kt

```diff
@@ -1,85 +1,63 @@
+    var selectedFiles by mutableStateOf(listOf<RecordingItemData>())
 
-    val recordingItems = mutableStateListOf<RecordingItemData>()
+    private var sortOrder = SortOrder.DEFAULT
 
-    fun loadFiles(context: Context) {
+    var audioRecordingItems by mutableStateOf(listOf<RecordingItemData>())
+    var screenRecordingItems by mutableStateOf(listOf<RecordingItemData>())
+
+    init {
+        loadFiles()
+    }
+
+    fun loadFiles() {
         viewModelScope.launch {
-            withContext(Dispatchers.IO) {
-                files = getAvailableFiles(context).toMutableStateList()
-            }
-            loadRecordingItems(context)
+            audioRecordingItems = fileRepository.getAudioRecordingItems(sortOrder)
+            screenRecordingItems = fileRepository.getVideoRecordingItems(sortOrder)
         }
     }
 
-    fun sortRecordingItems(context: Context, sortOrder: SortOrder) {
-        files = when (sortOrder) {
-            SortOrder.ALPHABETIC -> files.sortedBy { it.name }
-            SortOrder.ALPHABETIC_REV -> files.sortedByDescending { it.name }
-            SortOrder.SIZE_REV -> files.sortedBy {
-                context.contentResolver.query(it.uri, null, null, null, null)?.use { cursor ->
-                    cursor.getLong(cursor.getColumnIndex(OpenableColumns.SIZE).absoluteValue)
-                }
-            }
-
-            SortOrder.SIZE -> files.sortedByDescending {
-                context.contentResolver.query(it.uri, null, null, null, null)?.use { cursor ->
-                    cursor.getLong(cursor.getColumnIndex(OpenableColumns.SIZE).absoluteValue)
-                }
-            }
-        }.toMutableStateList()
-        loadRecordingItems(context)
+    fun sortItems(newSortOrder: SortOrder) {
+        if (newSortOrder == sortOrder) return
+        sortOrder = newSortOrder
+        loadFiles()
     }
 
-    private fun loadRecordingItems(context: Context) {
+    fun deleteFiles() {
         viewModelScope.launch {
-            recordingItems.clear()
-            files.forEach { file ->
-                if (file.name.orEmpty()
-                        .endsWith("mp4" /* Currently there are only mp4 video files*/)
-                ) {
-                    val thumbnail =
-                        MediaMetadataRetriever().apply {
-                            setDataSource(
-                                context,
-                                file.uri
-                            )
-                        }.frameAtTime
-                    recordingItems.add(RecordingItemData(file, RecorderType.VIDEO, thumbnail))
-                } else {
-                    recordingItems.add(RecordingItemData(file, RecorderType.AUDIO))
-                }
+            if (selectedFiles.isEmpty()) {
+                fileRepository.deleteAllFiles()
+                loadFiles()
+                return@launch
             }
+            fileRepository.deleteSelectedFiles(selectedFiles.map { it.recordingFile })
```

Comment:
Found an issue in my code.
I forgot to clear the selectedFiles list after deleting the selected files.

### Steps to reproduce bug
- Select a few Items and delete them
- After the deletion is complete, click the delete all button
- *Nothing happens*

Delete all button only works when there are no items selected (when the selectedFiles is empty).
But in here since I forgot to clear the list, the delete all button does not work.

Reply:
fixed in https://github.com/Bnyro/RecordYou/commit/ecf10af2d8f162ac53772a9ea52f915a2653a98f

Reply:
thanks for spotting <3